controllers/argocd/argocd_controller.go

```diff
@@ -107,22 +124,24 @@ func (r *ReconcileArgoCD) Reconcile(ctx context.Context, request ctrl.Request) (
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
-			return reconcile.Result{}, nil
+			return reconcile.Result{}, argocd, nil
 		}
 		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
+		return reconcile.Result{}, argocd, err
 	}
 
 	// Fetch labelSelector from r.LabelSelector (command-line option)
 	labelSelector, err := labels.Parse(r.LabelSelector)
 	if err != nil {
-		reqLogger.Info(fmt.Sprintf("error parsing the labelSelector '%s'.", labelSelector))
-		return reconcile.Result{}, err
+		message := fmt.Sprintf("error parsing the labelSelector '%s'.", labelSelector)
+		reqLogger.Info(message)
+		return reconcile.Result{}, argocd, fmt.Errorf(message, " error: %w", err)
```

Comment:
Can you double check the use of `Errorf` in this PR?

AFAICT `fmt.Errorf(message, " error: %w", err)` should be `fmt.Errorf("%s error: %w", message, err)`, and I see other instance as well